src/deepsparse/v2/text_generation/prep_for_generation.py

```diff
@@ -107,13 +66,13 @@ def run(
             logits_shape=prompt_logits[0, -1, :].shape,
             deterministic=not generation_config.do_sample,
             sampling_temperature=generation_config.temperature,
-            tokens=tokens,
+            tokens=copy.copy(tokens),
```

Comment:
why is that needed?

Reply:
it doesn't really matter but the TokenGenerator will update our local tokens list internally which isn't great for readability